megamek/src/megamek/server/Server.java

```diff
@@ -9956,6 +10022,64 @@ else if (bldgExited.equals(bldgEntered)
             }
         }
     }
+    
+    /**
+     * Updates the position of any towed trailers.
+     *
+     * @param tractor    The Entity that is moving
+     * @param trailer    The current trailer being updated
+     */
+    private void processTrailerMovement(Entity tractor, Entity trailer, ArrayList<Coords> trainPath) {
+        double trailerPositionOffset = 0;
+        int stepNumber = 0;
+        Coords trailerPos = null;
+        trailerPositionOffset = (tractor.getAllTowedUnits().indexOf(trailer) + 2); //Offset so we get the right position index
```

Comment:
Do you need to check for the case where `indexOf` returns -1 because `trailer` is not being towed by `tractor`?

Reply:
I *think* I'm safe.  This method should only get called when tractor is towing something (getAllTowedUnits().size() > 0.   It's intended that only tractors should be able to get here and not other trailers to which (trailer) is actually attached